Source/WebCore/platform/graphics/GraphicsStyle.h

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2022 Apple Inc.  All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "Color.h"
+#include "FloatSize.h"
+#include <wtf/Vector.h>
+
+namespace WTF {
+class TextStream;
+}
+
+namespace WebCore {
+
+struct GraphicsDropShadow {
+    FloatSize offset;
+    FloatSize radius;
+    Color color;
+    
+    template<class Encoder> void encode(Encoder&) const;
+    template<class Decoder> static std::optional<GraphicsDropShadow> decode(Decoder&);
+};
+
+inline bool operator==(const GraphicsDropShadow& a, const GraphicsDropShadow& b)
+{
+    return a.offset == b.offset && a.radius == b.radius && a.color == b.color;
+}
+
+struct GraphicsGaussianBlur {
+    FloatSize radius;
+
+    template<class Encoder> void encode(Encoder&) const;
+    template<class Decoder> static std::optional<GraphicsGaussianBlur> decode(Decoder&);
+};
+
+inline bool operator==(const GraphicsGaussianBlur& a, const GraphicsGaussianBlur& b)
+{
+    return a.radius == b.radius;
+}
+
+struct GraphicsColorMatrix {
+    Vector<float> values;
```

Comment:
Will this always be 20 values? If so, can we make this `std::array<float, 20>` instead?

Reply:
Also could ColorMatrix<4,5> class :).

Reply:
`std::array<float, 20>` was used. I had to add `operator<<` for `TextStream` when the argument is `std::array`.